File: MiniDOMOffice/Office/Lavoro/Curricula/CurriculumCursor.vb
Imports minidom.Anagrafica
Imports minidom.Databases
Imports minidom.Sistema


Partial Class Office

    ''' <summary>
    ''' Cursore sulla tabella dei curricula
    ''' </summary>
    ''' <remarks></remarks>
    Public Class CurriculumCursor
        Inherits DBObjectCursorPO(Of Curriculum)

        Private m_DataPresentazione As New CCursorField(Of Date)("DataPresentazione")
        Private m_IDPersona As New CCursorField(Of Integer)("IDPersona")
        Private m_NomePersona As New CCursorFieldObj(Of String)("NomePersona")
        Private m_Descrizione As New CCursorFieldObj(Of String)("Descrizione")
        Private m_IDAllegato As New CCursorField(Of Integer)("IDAllegato")

        Public Sub New()
        End Sub

        Public ReadOnly Property DataPresentazione As CCursorField(Of Date)
            Get
                Return Me.m_DataPresentazione
            End Get
        End Property

        Public ReadOnly Property IDPersona As CCursorField(Of Integer)
            Get
                Return Me.m_IDPersona
            End Get
        End Property

        Public ReadOnly Property NomePersona As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomePersona
            End Get
        End Property

        Public ReadOnly Property Descrizione As CCursorFieldObj(Of String)
            Get
                Return Me.m_Descrizione
            End Get
        End Property

        Public ReadOnly Property IDAllegato As CCursorField(Of Integer)
            Get
                Return Me.m_IDAllegato
            End Get
        End Property

        Protected Overrides Function GetConnection() As CDBConnection
            Return Office.Database
        End Function

        Public Overrides Function GetTableName() As String
            Return "tbl_OfficeCurricula"
        End Function

        Protected Overrides Function GetModule() As CModule
            Return Office.Curricula.Module
        End Function

    End Class


End Class

File: MiniDOMOffice/Office/Lavoro/Curricula/CurriculumCursor.cs
using System;
using System.Collections;
using System.Collections.Generic;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using minidom;
using minidom.repositories;
using static minidom.Sistema;
using static minidom.Anagrafica;
using static minidom.Office;

namespace minidom
{
    public partial class Office
    {

        /// <summary>
        /// Cursore di <see cref="Curriculum"/>
        /// </summary>
        /// <remarks></remarks>
        [Serializable]
        public class CurriculumCursor 
            : minidom.Databases.DBObjectCursorPO<Curriculum>
        {
            private DBCursorField<DateTime> m_DataPresentazione = new DBCursorField<DateTime>("DataPresentazione");
            private DBCursorField<int> m_IDPersona = new DBCursorField<int>("IDPersona");
            private DBCursorStringField m_NomePersona = new DBCursorStringField("NomePersona");
            private DBCursorStringField m_Descrizione = new DBCursorStringField("Descrizione");
            private DBCursorField<int> m_IDAllegato = new DBCursorField<int>("IDAllegato");

            /// <summary>
            /// Costruttore
            /// </summary>
            public CurriculumCursor()
            {
            }

            /// <summary>
            /// DataPresentazione
            /// </summary>
            public DBCursorField<DateTime> DataPresentazione
            {
                get
                {
                    return m_DataPresentazione;
                }
            }

            /// <summary>
            /// IDPersona
            /// </summary>
            public DBCursorField<int> IDPersona
            {
                get
                {
                    return m_IDPersona;
                }
            }

            /// <summary>
            /// NomePersona
            /// </summary>
            public DBCursorStringField NomePersona
            {
                get
                {
                    return m_NomePersona;
                }
            }

            /// <summary>
            /// Descrizione
            /// </summary>
            public DBCursorStringField Descrizione
            {
                get
                {
                    return m_Descrizione;
                }
            }

            /// <summary>
            /// IDAllegato
            /// </summary>
            public DBCursorField<int> IDAllegato
            {
                get
                {
                    return m_IDAllegato;
                }
            }
  
            /// <summary>
            /// Repository
            /// </summary>
            /// <returns></returns>
            protected override CModulesClass GetModule()
            {
                return minidom.Office.Curricula;
            }
        }
    }
}